common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -284,4 +284,15 @@ public static UTF8String getCollationKey(UTF8String input, int collationId) {
     }
   }
 
+  public static byte[] getCollationKeyBytes(UTF8String input, int collationId) {
+    Collation collation = fetchCollation(collationId);
+    if (collation.supportsBinaryEquality) {
+      return input.getBytes();
```

Comment:
this means memory copy. Shall we make this function return `UTF8String`? Then we don't need to call `getBytes` here, and it's pretty easy to create `UTF8String` from `byte[]`.

Reply:
the idea is we short-circuit for binary collation anyway (this branch is never executed in the current implementation)

the only reason why this "if" is included here is because we made is part of CollationFactory public interface (if someone else calls this with collationId = 0 for example)